Decouple model e2e tests

guangy10 opened the pull request:
Differential Revision: D49622665

huydhn reviewed (approved):
LGTM! There are only few minor typos

guangy10 committed:
Decouple model e2e tests (#487)

Summary:

We should not couple e2e tests with different recipes *(e.g. run with portable kernels, run with xnnpack options)* in a single job. Addressing timeout issue and improving CI stability is one of the reasons, however, even w/o timeout we still want to do it for:
  1. Maximal parallelism to reduce waiting time
  2. Flexibility to enable/disable an e2e test for a given recipe, e.g. disable slow running e2e test with portable kernels but keep the xnnpack e2e tests enabled, or if there are issue with a flavor
  3. Cleaner CI job names to describe what it does

Of course this is not real fix of the timeout issue we're facing, the complete fix will include 1) introducing release build for both cmake and buck2 and 2) fixing the slow portable kernels from the root. And both are far away and not possible before MVP.

For MVP as we have more and more models enabled with XNNPACK as a part of the out-of-box experience, our CI is facing extra stress. See the [ci dashboard](https://hud.pytorch.org/hud/pytorch/executorch/main). The major motivation in this diff is to ensure the healthiness of enabled models as well as our CI.

Reviewed By: huydhn, kirklandsign

Differential Revision: D49622665
